src/audio/mixer.c

```diff
@@ -174,25 +174,21 @@ static int mixer_params(struct comp_dev *dev,
 {
 	struct sof_ipc_comp_config *config = COMP_GET_CONFIG(dev);
 	struct comp_buffer *sinkb;
-	struct comp_buffer *sourceb;
 	uint32_t period_bytes;
 
 	trace_mixer_with_ids(dev, "mixer_params()");
 
-	sourceb = list_first_item(&dev->bsource_list, struct comp_buffer,
-				  sink_list);
+	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer,
+							source_list);
```

Comment:
@bkokoszx just wondering if this should be based on the stream direction instead of always using the sink buffer?

Reply:
@ranj063 
At the moment I think we should use sink buffer for mixer, because it has one sink (some of source buffers can be unset and as a result ```period_bytes``` can be set to 0 causing errors).